### PR TITLE
Fix missing loactaion in Encounter.hbs and validate the data in test

### PR DESCRIPTION
**Pull request by QuanWanxx**
This PR is mean to fix the "location(reference)" property missing in Encounter.hbs issue.   
Update the template and validate the generated data in test folder

### CODE REVIEW

**Comment by ACMoretxj** [comment]
@ranvijaykumar Ranvijay, I'm wondering now that we have the reference to a location resource, should we also add a location resource statement(like `{{>Resources/Location.hbs PV1-3=PV1-3...blahblah.....}}`) inside `ADT_A01.hbs`?

**Comment by ranvijaykumar** [comment]
> @ranvijaykumar Ranvijay, I'm wondering now that we have the reference to a location resource, should we also add a location resource statement(like `{{>Resources/Location.hbs PV1-3=PV1-3...blahblah.....}}`) inside `ADT_A01.hbs`?

Yes, we should.

**Comment by QuanWanxx** [comment]
> > @ranvijaykumar Ranvijay, I'm wondering now that we have the reference to a location resource, should we also add a location resource statement(like `{{>Resources/Location.hbs PV1-3=PV1-3...blahblah.....}}`) inside `ADT_A01.hbs`?
> 
> Yes, we should.

Hi, I discussed with @ACMoretxj , the current Location.hbs contains all possible data entities may represent a location value
image

So if we add **{{>Resources/Location.hbs PV1=PV1 ...blahblah.....}}** inside the ADT_AO1.hbs, the different reference path will actually point to same resource instance in generated data. And if we change this template, may lead to other issues related to Location.hbs.

I thought maybe we can just not add  **{{>Resources/Location.hbs PV1=PV1 ...blahblah.....}}** inside the ADT_AO1.hbs, and merge this branch to fix the missing location reference bug (maybe there are other this kind of process operation exist), and considering add the reference instance in the future?

File: src/templates/hl7v2/Resources/Encounter.hbs [hunk 1]
```diff
@@ -21,11 +21,21 @@
...
+                    {
+                        "reference": "Location/{{generateUUID PV1-3}}",
+                    }
+                {{/if}}
             },
             {
                 {{#if PV1-6}},
                     "status":"completed",
-                {{/if}},
+                    "location":
```
Comment (ACMoretxj) [inline src/templates/hl7v2/Resources/Encounter.hbs:34]: Maybe there should also have a "{{#if PV1-6}}," statement for consistency.

File: src/templates/hl7v2/Resources/Encounter.hbs [hunk 2]
```diff
@@ -21,11 +21,21 @@
...
+                    {
+                        "reference": "Location/{{generateUUID PV1-6}}",
```
Comment (BoyaWu10) [inline src/templates/hl7v2/Resources/Encounter.hbs:36]: We can update `Location`  to take `PL` data type as input. In this way PV1-3 and PV1-6 can generate a location each.

**Review by ACMoretxj — APPROVED** [review]

**Comment by BoyaWu10** [comment]
@ACMoretxj There are some changes in partial templates (e.g., Encounter, PractitionerRole). Could you please double check that these changes could be fixed in generation tool as well?

**Comment by ACMoretxj** [comment]
> @ACMoretxj There are some changes in partial templates (e.g., Encounter, PractitionerRole). Could you please double check that these changes could be fixed in generation tool as well?

Yes, all the changes of partial templates are all able to be fixed in the generation tool.

**Comment by BoyaWu10** [comment]
> > @ACMoretxj There are some changes in partial templates (e.g., Encounter, PractitionerRole). Could you please double check that these changes could be fixed in generation tool as well?
> 
> Yes, all the changes of partial templates are all able to be fixed in the generation tool.

Got it. Thanks!

**Review by BoyaWu10 — DISMISSED** [review]

**Review by ACMoretxj — APPROVED** [review]

**Review by BoyaWu10 — APPROVED** [review]

**Pull request merged**